process/block/baseProcess.go

```diff
@@ -801,6 +793,50 @@ func (bp *baseProcessor) getLastCrossNotarizedHeadersForShard(shardID uint32) *b
 	return headerInfo
 }
 
+func (bp *baseProcessor) getLastSelfNotarizedHeaders() []bootstrapStorage.BootstrapHeaderInfo {
```

Comment:
why not return list of pointers of bootstrapheaderInfo. []*bootstrapStorage.BootstrapHeaderInfo

Reply:
Because the result should be set in BootstrapData structure from bootstrapData.pb.go which hold a list of objects not a list of pointers. This would be then marshalized and put it in bootstorer on disk.

Reply:
// BootstrapData is used to store information that are needed for bootstrap
type BootstrapData struct {
	LastHeader                 BootstrapHeaderInfo     `protobuf:"bytes,1,opt,name=LastHeader,proto3" json:"LastHeader"`
	LastCrossNotarizedHeaders  []BootstrapHeaderInfo   `protobuf:"bytes,2,rep,name=LastCrossNotarizedHeaders,proto3" json:"LastCrossNotarizedHeaders"`
	LastSelfNotarizedHeaders   []BootstrapHeaderInfo   `protobuf:"bytes,7,rep,name=LastSelfNotarizedHeaders,proto3" json:"LastSelfNotarizedHeaders"`
	ProcessedMiniBlocks        []MiniBlocksInMeta      `protobuf:"bytes,4,rep,name=ProcessedMiniBlocks,proto3" json:"ProcessedMiniBlocks"`
	PendingMiniBlocks          []PendingMiniBlocksInfo `protobuf:"bytes,8,rep,name=PendingMiniBlocks,proto3" json:"PendingMiniBlocks"`
	NodesCoordinatorConfigKey  []byte                  `protobuf:"bytes,9,opt,name=NodesCoordinatorConfigKey,proto3" json:"NodesCoordinatorConfigKey,omitempty"`
	EpochStartTriggerConfigKey []byte                  `protobuf:"bytes,10,opt,name=EpochStartTriggerConfigKey,proto3" json:"EpochStartTriggerConfigKey,omitempty"`
	HighestFinalBlockNonce     uint64                  `protobuf:"varint,5,opt,name=HighestFinalBlockNonce,proto3" json:"HighestFinalBlockNonce,omitempty"`
	LastRound                  int64                   `protobuf:"varint,6,opt,name=LastRound,proto3" json:"LastRound,omitempty"`
}